File: is_palindrome/is_palindrome.java
public class PalindromeChecker {
  public static boolean isPalindrome(String str) {
    int firstIdx = 0;
    int lastIdx = str.length() - 1;
    while (firstIdx <= lastIdx) {
      if (str.charAt(firstIdx) != str.charAt(lastIdx)) {
        return false;
      }
      firstIdx++;
      lastIdx--;
    }
    return true;
  }

  public static void main(String[] args) {
    System.out.println(isPalindrome("aba"));
  }
}

File: is_palindrome/is_palindrome.py
def is_palindrome(string):
  first_letter_idx = 0
  last_letter_idx = len(string) - 1
  while first_letter_idx <= last_letter_idx:
    if string[first_letter_idx] != string[last_letter_idx]:
      return False
    first_letter_idx += 1
    last_letter_idx -= 1
  return True

print(is_palindrome('aba'))
